docs/rfc/block-registration.md

```diff
@@ -331,6 +331,36 @@ Block type editor style definition. It will only be enqueued in the context of t
 
 Block type frontend style definition. It will be enqueued both in the editor and when viewing the content on the front of the site.
 
+### Provides Context
+
+* Type: `object`
+* Optional
+* Localized: No
+* Property: `providesContext`
+
+Context provided for available access by descendants of blocks of this type, in the form of an object which maps a context name to one of the block's own attribute.
+
+See [the block context documentation](/docs/designers-developers/developers/block-api/block-context.md) for more details.
+
+```json
+{ "providesContext": { "my-plugin/recordId": "recordId" } }
+```
+
+### Context
+
+* Type: `string[]`
+* Optional
+* Localized: No
+* Property: `context`
```

Comment:
I don't like this naming. Context is already in use in the REST API and it reusing this naming is super confusing. 
If I make a request to `/wp/v2/block-type/?context=edit` what do I get. The request in an edit context and block files by context === edit. Confusing. Can we use another name here?

Reply:
How about the naming `block_context`?

Reply:
Interesting, it's a completely different concept from what REST API uses. It didn't occur to me earlier that the name might be too general, it's more like:
- `dataContext` or `externalAttributes`
- `providesAttributes`

I'm curious to hear how @aduth views it.

Reply:
I'd consider "context" a fairly generic term of the likes of "type", and the relationship between blocks and the REST API is incidental enough that—outside this specific collision—I can't imagine there'd be a high likelihood for confusion of the two.

Conversely, as I understand from some of early discussions like in #19685, there's an intentional overlap of the name it and the [concept of Context in React](https://reactjs.org/docs/context.html) which it mimics (and uses for implementation in the client).

I'm not especially attached to the name, but it wasn't really the intent of the pull request to be _introducing_ the concept, moreso to bring this document into alignment with what had already been implemented across discussions and iterations of #19685, #19572, and #21467.

There's some interesting merit to thinking about this in terms of "attributes provided from some ancestor" and potentially naming around that, though even that may restrict some options for how context could expand in the future (see https://github.com/WordPress/gutenberg/issues/19685#issuecomment-601341635, https://github.com/WordPress/gutenberg/issues/19685#issuecomment-603848589).

Since there's far more impact to a decision than what's affected here, I'd rather weigh it in its own dedicated space (an issue or [editor meeting](https://make.wordpress.org/meetings/) discussion).

Reply:
Context is already in use in core and to add this here would be really confusing. Please can we just rename it?